contrib/src/main/java/gov/nasa/jpl/aerie/contrib/streamline/modeling/polynomial/Polynomial.java

```diff
@@ -0,0 +1,308 @@
+      //   to do a more thorough but still efficient search?
+    } catch (ArithmeticException e) {
+      // If we overflowed looking for a bracketing range, it effectively never transitions.
+      return NEVER;
+    }
+
+    // Do a binary search to find the exact transition time
+    while (end.longerThan(start.plus(EPSILON))) {
+      Duration midpoint = start.plus(end).dividedBy(2);
+      if (expires.test(midpoint)) {
+        end = midpoint;
+      } else {
+        start = midpoint;
+      }
+    }
+    return Expiry.at(end);
+  }
+
+  private Expiring<Discrete<Boolean>> greaterThan(Polynomial other, boolean strict) {
+    BiPredicate<Double, Double> comp = strict ? (x, y) -> x > y : (x, y) -> x >= y;
+    boolean result = comp.test(this.extract(), other.extract());
+    var expiry = this.subtract(other).findExpiryNearRoot(
+        t -> comp.test(this.evaluate(t), other.evaluate(t)) != result);
+    return expiring(discrete(result), expiry);
+  }
+
+  public Expiring<Discrete<Boolean>> greaterThan(Polynomial other) {
+    return greaterThan(other, true);
+  }
+
+  public Expiring<Discrete<Boolean>> greaterThanOrEquals(Polynomial other) {
+    return greaterThan(other, false);
+  }
+
+  public Expiring<Discrete<Boolean>> lessThan(Polynomial other) {
+    return other.greaterThan(this);
+  }
+
+  public Expiring<Discrete<Boolean>> lessThanOrEquals(Polynomial other) {
+    return other.greaterThanOrEquals(this);
+  }
+
+  private boolean dominates$(Polynomial other) {
+    for (int i = 0; i <= Math.max(this.degree(), other.degree()); ++i) {
+      if (this.getCoefficient(i) > other.getCoefficient(i)) return true;
+      if (this.getCoefficient(i) < other.getCoefficient(i)) return false;
+    }
+    // Equal, so either answer is correct
+    return true;
+  }
+
+  private Expiring<Discrete<Boolean>> dominates(Polynomial other) {
+    boolean result = this.dominates$(other);
+    var expiry = this.subtract(other).findExpiryNearRoot(t -> this.step(t).dominates$(other.step(t)) != result);
+    return expiring(discrete(result), expiry);
+  }
+
+  public Expiring<Polynomial> min(Polynomial other) {
+    return ExpiringMonad.map(this.dominates(other), d -> d.extract() ? other : this);
+  }
+
+  public Expiring<Polynomial> max(Polynomial other) {
+    return ExpiringMonad.map(this.dominates(other), d -> d.extract() ? this : other);
+  }
+
+  /**
+   * Finds all roots of this function in the future
+   */
+  private Stream<Duration> findFutureRoots() {
+    // If this polynomial can never have a root, fail immediately
+    if (this.isNonFinite() || this.isConstant()) {
+      return Stream.empty();
+    }
+
+    // Defining epsilon keeps the Laguerre solver fast and stable for poorly-behaved polynomials.
+    final double epsilon = 2 * Arrays.stream(coefficients).map(Math::ulp).max().orElseThrow();
+    final Complex[] solutions = new LaguerreSolver(0, ABSOLUTE_ACCURACY_FOR_DURATIONS, epsilon)
```

Comment:
The Laguerre solver can enter an infinite loop with polynomial coefficients [Double.MAX_VALUE, -1.0].  This is currently a problem for the SRL cruise model.  Inside the solver, the variables inside the `while(true)` loop are mostly (if not all) NaNs at some point, disallowing it to find and return a root.

Reply:
A bandaid for this case could be to solve linear Polynomials in this function directly and only use Laguerre for higher order polynomials.

Reply:
I remember coming across a different case where this or some other root-finding solver used in the `streamline` package is slow to converge for coefficients close to 0.0.  I forgot what we did, if anything, to avoid it, so that may be another ongoing issue.  I remember some ideas, but I expect we just modified the mission model to avoid it.  I'll try that for this case, too.